test/unit/spec/nbextensions/bulkImportCell/tabs/xsvGenerator-spec.js

```diff
@@ -0,0 +1,369 @@
+                            ['fastq_rev_staging_file_name', 'Reverse/right FASTQ file path'],
+                            ['name', 'Reads object name'],
+                            ['read_orientation_outward', 'Reads orientation outward'],
+                        ],
+                        data: [
+                            {
+                                fastq_fwd_staging_file_name: null,
+                                fastq_rev_staging_file_name: null,
+                                name: null,
+                                read_orientation_outward: 0,
+                            },
+                            {
+                                fastq_fwd_staging_file_name: null,
+                                fastq_rev_staging_file_name: null,
+                                name: null,
+                                read_orientation_outward: 0,
+                            },
+                        ],
+                    },
+                    sra_reads: {
+                        order_and_display: [
+                            ['sra_staging_file_name', 'SRA file path'],
+                            ['sequencing_tech', 'Sequencing technology'],
+                            ['name', 'Reads object name'],
+                            ['read_orientation_outward', 'Reads orientation outward'],
+                        ],
+                        data: [
+                            {
+                                name: 'sample_sra_reads',
+                                sra_staging_file_name: 'sample_sra',
+                                read_orientation_outward: 0,
+                                sequencing_tech: 'Illumina',
+                            },
+                        ],
+                    },
+                };
+
+                it('generates params with a single input', function () {
+                    const output = this.xsvGen.generateRequest({
+                        output_file_type,
+                        output_directory,
+                        types: ['sra_reads'],
+                    });
+                    expect(output).toEqual({
+                        output_file_type,
+                        output_directory,
+                        types: {
+                            sra_reads: expectedOutput.sra_reads,
+                        },
+                    });
+                });
+
+                it('generates params with several inputs', function () {
+                    const output = this.xsvGen.generateRequest({
+                        output_directory,
+                        output_file_type,
+                        types: Object.keys(params),
+                    });
+                    expect(output).toEqual({
+                        types: expectedOutput,
+                        output_directory,
+                        output_file_type,
+                    });
+                });
+            });
+
+            describe('responses', () => {
```

Comment:
testing the response to the staging service query (even though the staging service query itself has not been hooked up)

Reply:
OK, it has been hooked up now